java/ingest/ingest-batcher-submitterv2/src/test/java/sleeper/ingest/batcher/submitterv2/IngestBatcherSubmitterLambdaIT.java

```diff
@@ -119,14 +164,68 @@ void shouldStoreOneFileInDirectory() {
                     "}";
 
             // When
-            lambda.handleMessage(json, RECEIVED_TIME);
+            lambda().handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(batcherStore().getAllFilesNewestFirst())
+                    .containsExactly(
+                            fileRequest(testBucket + "/test-directory/test-file-1.parquet"));
+        }
+
+        @Test
+        void shouldStoreFilesInDirectoryWhenSubmittedByFullName() {
+            // Given
+            uploadFileToS3("test-directory/test-file-1.parquet");
+            String json = "{" +
+                    "\"files\":[\"" + testBucket + "/test-directory/test-file-1.parquet\"]," +
+                    "\"tableName\":\"test-table\"" +
+                    "}";
+
+            // When
+            lambda().handleMessage(json, RECEIVED_TIME);
 
             // Then
-            assertThat(store.getAllFilesNewestFirst())
+            assertThat(batcherStore().getAllFilesNewestFirst())
                     .containsExactly(
                             fileRequest(testBucket + "/test-directory/test-file-1.parquet"));
         }
 
+        @Test
+        void shouldStoreFilesInDeepDirectory() {
+            // Given
+            uploadFileToS3("test-directory/another-test-directory/test-file-1.parquet");
+            String json = "{" +
+                    "\"files\":[\"" + testBucket + "/test-directory/another-test-directory\"]," +
+                    "\"tableName\":\"test-table\"" +
+                    "}";
+
+            // When
+            lambda().handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(batcherStore().getAllFilesNewestFirst())
+                    .containsExactly(
+                            fileRequest(testBucket + "/test-directory/another-test-directory/test-file-1.parquet"));
+        }
+
+        @Test
+        void shouldStoreFilesInDeepDirectoryEvenWhenCalledByFileName() {
+            // Given
+            uploadFileToS3("test-directory/another-test-directory/test-file-1.parquet");
+            String json = "{" +
+                    "\"files\":[\"" + testBucket + "/test-directory/another-test-directory/test-file-1.parquet\"]," +
+                    "\"tableName\":\"test-table\"" +
+                    "}";
```

Comment:
I think this test belongs in StoreSingleFile. This nested test class is for submitting a directory rather than a file.